Saurav

sauravbose opened the pull request:
TuRF implementation completed.

sauravbose committed:
TuRF detailed prints with header settings

sauravbose committed:
All core algos removed comments

sauravbose committed:
Modified docs

sauravbose committed:
Updated docstrings

weixuanfu commented on tests.py:
Maybe you need increase `step` herein to speed up the unit test, which is why the builds in python 3.6 were failed in ci.

weixuanfu commented on skrebate/turf.py:
Please specify the class/func names when import them instead of using `*`.

weixuanfu commented on skrebate/turf.py:
I think the situation when step is `int` maybe missing here.

weixuanfu commented on skrebate/turf.py:
please make a note about this condition in docstring.

sauravbose committed:
multiclass

sauravbose committed:
multiclass

sauravbose committed:
made attr a class variable and added multiclass capability for all core algos

sauravbose committed:
Updated far scoring in multiclass case

Commit:
Added clarifying comments to fit() in relieff

Commit:
Moved deletion of distance array to just after calculation of feature
importance scores. Might as well do this as soon as possible.

Commit:
Added clarifying comments to the rest of relieff.py and
scoring_utils.py.  Inserted commented out code to fix the distance array
calculation when missing data is present. Not implemented yet.

Commit:
Fixed 'get_row_missing() in scoring_utils.py so that distance array
calculation is properly normalized by the number of uniquely missing
features in comparing instance pair distances. This way distance is
computed agnostically with respect to missing values.  The previous
implementation would make instances with more missing values appear
closer to one another.

Commit:
Partial fix to proper normalization of continuous value range in
distance calculation within get_row_missing().  Added 'cmins' variable
passed forward for subtraction of minimum value.  Some modifications
still needed to complete this fix.

ryanurbs committed:
Added more comments. Identified an issue in relieff, find_neighbors().
It appears it it only set up to handle binary endpoints correctly, not
multiclass or continuous valued endpoints. This problem does not exist
for the other 4 core algorithms (SURF, SURF*, MultiSURF, and MultiSURF*)

ryanurbs committed:
Minor comment and code rearrangement

ryanurbs committed:
added further comments to the code for clarification and to identify
possible changes

Commit:
Added Saurav's fix for nearest neighbor selection of multiclass and
binary class endpoints.

ryanurbs committed:
Added further comments.  Made absoulte value fix to Relieff's continuous
endpoint nearest neighbor determination.  Also got rid of mmdiff in
scoring_utils for discrete endpoints (only needed for continuous
endpoints.) Identified a problem in 'compute_score'.  For #far score
contributions, continuous valued features should not check for equality
(this will lead to many scores being left out.)

ryanurbs committed:
Further comments. Got rid of _X prenormalization (this changed the
original data array only for datasets with all continuous values) now
prenormalization is run on 'xc' for such data. 
Also Fixed scoring update for any continuous features (no prior feature
equivalence check.  This causes definitely problems for data with
continuous features particularly in MultiSURF*. So far only fixed for
binary endpoints.
*fixed normalization for binary endpoint scoring.  Added normalization
by 'n' (number of training instances) this doesn't appear to be in here
anywhere for any of the Relief methods.

ryanurbs committed:
Completed fixes to 'compute_score' to fixe continuous feature scoring
issues as well as normalization fixes and update to be more relatable to
the rebate papers.

ryanurbs committed:
Added remaining clarifying comments to individual RBA module files.

ryanurbs committed:
minor comment addition

ryanurbs committed:
Manually added weixuan fu's bug fixes for scoring_units about data_len
and mmdiff.  I also changed the use of abs value continuous feature
difference and mmdiff normalization so it's only called when a
continuous feature is present rather than for any feature regardless.

ryanurbs committed:
Made fix pointed out by Weixuan regarding special case when count_hit or
count_miss happens to be zero.

ryanurbs committed:
analysis example dataset file path (csv changed to tsv)

ryanurbs committed:
Fixed special case where both count hit and miss are zero.  This could
happen when there are very few features and instances.

ryanurbs committed:
in tests.py, changed all TuRF step sizes from 0.1 to 0.4 to speed up
unit testing

ryanurbs committed:
Fixed some errors in my ramp function implementation causing additional
test errors.

ryanurbs committed:
Minor fix to ramp function implementation to fix error in integrated
testing.
